### PR TITLE
decomp: add a bunch of enemy files

### PR DESCRIPTION
**Pull request by Hat-Kid**
Adds the following files:

- `amphibian`
- `centurion`
- `ginsu`
- `grenadier`
- `hopper`
- `metalmonk`
- `monster-frog`
- `predator-graph`
- `predator-h`
- `predator`
- `rapid-gunner`
- `rhino`
- `rhino-wall`
- `tomb-baby-spider`

Also adds the DGOs for the following levels:

- Mar's Tomb
- Mountain Temple
- Drill Platform
- Sacred Site (Sage Hut)

Manual patches:

- The decompiler emits `(b! #t cfg-17)` in `(trans hostile hopper)` without putting a `(label cfg-17)` anywhere
- Added cast to `art-joint-anim` in `(code broken rhino-wall)` and `(code hit rhino-wall)`

Other notes:

- `amphibian` seems to occasionally crash when using its tongue attack. Haven't investigated this yet
- `ginsu` crashes after being killed somewhere in the `deactivate` method, possibly because of its `part-spawner`
- Predators aren't spawning in the forest hunt mission, not sure if `forest-obs` might be needed for that or if it's something else
- The `rhino-wall` STR animation seems to load (albeit a bit broken due to missing VAG stream playback), but causes the viewport to shrink as soon as it's played
- I added `pegasus::74` to the `event_handler_hack` bool in `variable_naming.cpp` because I got a `none` cast after changing the return value of `enemy::74`

### CODE REVIEW

File: goal_src/jak2/levels/forest/predator.gc [hunk 1]
```diff
@@ -7,3 +7,1593 @@
...
+    )
+  :code (the-as (function none :behavior predator-manager) sleep-code)
+  )
+
+(defstate closed (predator-manager)
+  :virtual #t
+  :code (the-as (function none :behavior predator-manager) sleep-code)
+  )
+
+;; ERROR: function was not converted to expressions. Cannot decompile.
```
Comment (water111) [inline goal_src/jak2/levels/forest/predator.gc:1599]: maybe this is why they don't spawn?

File: goal_src/jak2/levels/ruins/rapid-gunner.gc [hunk 2]
```diff
@@ -7,3 +7,1396 @@
...
+         (sound-play "rgun-shot-hit")
+         )
+     )
+    (else
+      ((method-of-type nav-enemy enemy-method-74) obj arg0 arg1 arg2 arg3)
+      )
+    )
+  )
+
+;; ERROR: function was not converted to expressions. Cannot decompile.
```
Comment (water111) [inline goal_src/jak2/levels/ruins/rapid-gunner.gc:192]: another one here?

File: goal_src/jak2/levels/ruins/rapid-gunner.gc [hunk 3]
```diff
@@ -7,3 +7,1396 @@
...
+        (suspend)
+        (ja :num! (seek!))
+        )
+      )
+    #f
+    (none)
+    )
+  :post (behavior ()
+    (let* ((a0-0 self)
+           (t9-0 (method-of-object a0-0 rapid-gunner-method-185))
```
Comment (water111) [inline goal_src/jak2/levels/ruins/rapid-gunner.gc:638]: I think this method is missing a float argument (the `#x40800000` below)

**Review by water111 — CHANGES_REQUESTED** [review]

File: goal_src/jak2/levels/forest/predator.gc (hunk 1 above)
Comment (Hat-Kid) [reply goal_src/jak2/levels/forest/predator.gc]: Yup, I overlooked some function that were marked as asm. They spawn now, but crash in merc code, likely related to their fancy cloak effect which uses generic.

File: goal_src/jak2/levels/common/enemy/amphibian/amphibian.gc [hunk 4]
```diff
@@ -7,3 +7,1838 @@
...
+     )
+    (else
+      ((method-of-type nav-enemy enemy-method-74) obj arg0 arg1 arg2 arg3)
+      )
+    )
+  )
+
+(defun amphibian-joint-mod-callback ((arg0 cspace) (arg1 transformq))
+  (cspace<-parented-transformq-joint! arg0 arg1)
+  (let ((s3-0 (new 'stack-no-clear 'bone))
```
[the remark this answers is not in the document]
Comment (Hat-Kid) [reply goal_src/jak2/levels/common/enemy/amphibian/amphibian.gc]: Seems to have fixed the crash!

**Pull request merged**